src/routes/ViewAmendmentRoute.js

```diff
@@ -94,18 +100,55 @@ class ViewAmendmentsRoute extends React.Component {
       });
   }
 
+  handleDuplicate = (cloneableProperties) => {
+    const { history, match, stripes: { okapi } } = this.props;
+
+    return fetch(`${okapi.url}/licenses/amendments/${match.params.amendmentId}/clone`, {
+      method: 'POST',
+      headers: {
+        'X-Okapi-Tenant': okapi.tenant,
+        'X-Okapi-Token': okapi.token,
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(cloneableProperties),
+    }).then(response => {
+      if (response.ok) {
+        return response.text(); // Parse it as text
+      } else {
+        throw new Error(errorTypes.JSON_ERROR);
+      }
+    }).then(text => {
+      const data = JSON.parse(text); // Try to parse it as json
+      if (data.id) {
+        return Promise.resolve(history.push(`${this.urls.editAmendment(data.id)}`)); // Location search is already a part of urls.editAmendment
+      } else {
+        throw new Error(errorTypes.INVALID_JSON_ERROR); // when the json response body doesn't contain an id
+      }
+    }).catch(error => {
+      throw error;
+    });
+  }
+
   showDeleteConfirmationModal = () => this.setState({ showConfirmDelete: true });
 
   hideDeleteConfirmationModal = () => this.setState({ showConfirmDelete: false });
 
+  showDuplicateModal = () => {
+    this.setState({ showDuplicate: true });
+  }
+
+  closeDuplicateModal = () => {
+    this.setState({ showDuplicate: false });
+  }
+
   fetchIsPending = () => {
     return Object.values(this.props.resources)
       .filter(r => r && r.resource !== 'licenses')
       .some(r => r.isPending);
   }
 
   urls = {
-    editAmendment: this.props.stripes.hasPerm('ui-licenses.licenses.edit') && (() => `/licenses/${this.props.match.params.id}/amendments/${this.props.match.params.amendmentId}/edit${this.props.location.search}`),
+    editAmendment: this.props.stripes.hasPerm('ui-licenses.licenses.edit') && ((amendmentId = null) => `/licenses/${this.props.match.params.id}/amendments/${amendmentId || this.props.match.params.amendmentId}/edit${this.props.location.search}`),
```

Comment:
`amendmentId || this.props.match.params.amendmentId` -> Arent both being fetched from the same resource?

Reply:
No, the former is a passed param to that function, so that we can call editAmendment with a new id rather than the one in the match params.

This is because we edit from `licenses/{licensId}/amendments/{amendmentId1}` but want to end up editing licenses/{licensId}/amendments/{amendmentId2}

Reply:
From our conversation on slack, instead of having a conditional logic here, its better to pass in the actual id we expect to navigate to as the param to the `editAmendment` callback.

Reply:
Now changed so we always pass in amendmentId